samples/InputEcsSample/main.cpp

```diff
@@ -0,0 +1,162 @@
+            transforms.PushComponentUpdateInstruction(entity, newComponent);
+        }
+    });
+
+    scheduler.GetComponentCache().PrepAllBuffersForNextFrame(std::vector<EntityId>{});
+
+    NovelRT::Timing::StepTimer timer;
+
+    auto windowPtr = windowingProvider->GetWindowingDevice();
+    windowPtr->SetWindowTitle("ECS Input Test");
+
+    std::shared_ptr<NovelRT::Ecs::Input::InputSystem> inputSystem =
+        scheduler.GetRegisteredIEcsSystemAs<NovelRT::Ecs::Input::InputSystem>();
+
+    inputSystem->AddDefaultKBMMapping();
+
+    scheduler.RegisterSystem([&](auto delta, auto catalogue) {
+        ComponentView<NovelRT::Ecs::Input::InputEventComponent> events =
+            catalogue.template GetComponentView<NovelRT::Ecs::Input::InputEventComponent>();
+        ComponentView<TransformComponent> transforms = catalogue.template GetComponentView<TransformComponent>();
+        NovelRT::Maths::GeoVector2F scale = NovelRT::Maths::GeoVector2F::zero();
+        NovelRT::Maths::GeoVector2F move = NovelRT::Maths::GeoVector2F::zero();
+
+        for (auto [entity, input] : events)
+        {
+            // movement
+            if (inputSystem->GetMappingName(entity) == "Down" &&
+                (input.state == KeyState::KeyDown || input.state == KeyState::KeyDownHeld))
+            {
+                move += NovelRT::Maths::GeoVector2F{0.0f, -5.0f};
+            }
+            if (inputSystem->GetMappingName(entity) == "Up" &&
+                (input.state == KeyState::KeyDown || input.state == KeyState::KeyDownHeld))
+            {
+                move += NovelRT::Maths::GeoVector2F{0.0f, 5.0f};
+            }
+            if (inputSystem->GetMappingName(entity) == "Left" &&
+                (input.state == KeyState::KeyDown || input.state == KeyState::KeyDownHeld))
+            {
+                move += NovelRT::Maths::GeoVector2F{-5.0f, 0.0f};
+            }
+            if (inputSystem->GetMappingName(entity) == "Right" &&
+                (input.state == KeyState::KeyDown || input.state == KeyState::KeyDownHeld))
+            {
+                move += NovelRT::Maths::GeoVector2F{5.0f, 0.0f};
+            }
+
+            // scale
+            if (inputSystem->GetMappingName(entity) == "A" &&
+                (input.state == KeyState::KeyDown || input.state == KeyState::KeyDownHeld))
+            {
+                scale += NovelRT::Maths::GeoVector2F{0.2f, 0.2f};
+            }
+            if (inputSystem->GetMappingName(entity) == "B" &&
+                (input.state == KeyState::KeyDown || input.state == KeyState::KeyDownHeld))
+            {
+                scale += NovelRT::Maths::GeoVector2F{-0.2f, -0.2f};
+            }
+
+            if (inputSystem->GetMappingName(entity) == "LeftClick" &&
+                (input.state == KeyState::KeyDown || input.state == KeyState::KeyDownHeld))
+            {
+                logger.logInfo("Clicked at {}, {}", input.mousePositionX, input.mousePositionY);
+            }
+        }
```

Comment:
Can't we change this to purely direct access? This seems kinda weird.

Reply:
How would we do this? As it is, the mapping names (which define the mappings) are `std::string` and break the triviality of the component - as such, we need a way to have the system identify which input was found based on the entity ID the system initially gave it...the service has no idea about the IDs either.

I'm all ears on this one if you have suggestions

Reply:
The *service* has no idea of the IDs?

What is the service?

Reply:
Discussed in discord - I believe this should be resolved as of latest commit